tensorflow_addons/layers/adaptive_pooling.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Pooling layers with fixed size outputs"""
+
+import tensorflow as tf
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class AdaptiveAveragePooling2D(tf.keras.layers.Layer):
+    """Average Pooling with adaptive kernel size and strides.
+        Arguments:
+            h_bins: Output Height
+            w_bins: Output Width
+
+        Input shape:
+            - If `data_format='channels_last'`:
+                4D tensor with shape `(batch_size, height, width, features)`.
+            - If `data_format='channels_first'`:
+                4D tensor with shape `(batch_size, features, height, width)`.
+
+        Output shape:
+            - If `data_format='channels_last'`:
+                4D tensor with shape `(batch_size, h_bins, w_bins, features)`.
+            - If `data_format='channels_first'`:
+                4D tensor with shape `(batch_size, features, h_bins, w_bins)`.
+    """
+
+    def __init__(self, h_bins, w_bins, data_format="channels_last", **kwargs):
```

Comment:
I think instead of `h_bins` and `w_bins`, a better API design is to take a single argument say `output_size`. This matches the argument convention `kernel_size`, `pool_size` etc for pooling layers in core TF.